### ISSUE TITLE
Mixed proj_dtypes + F16-majority GGUF: export takes f32 path while codegen expects raw bytes

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/forge-llm#213]
**Severity:** P2 — silent layout corruption on a narrow edge case.

Found by codex review of v0.9.13 ([commit 619607c](https://github.com/sauravpanda/forge-llm/commit/619607c)).

## What's wrong

`cmd_export_weights_impl` gates the quantized export path on `config.dtype` being one of `{Q8_0, Q4_0, Q4_K, Q6_K}`:

https://github.com/sauravpanda/forge-llm/blob/main/crates/forgellm-cli/src/main.rs#L2562

But `config.dtype` is the file-level majority from `detect_gguf_dtype`, which falls back to `F16` when no quant has a strict majority. Meanwhile `config.proj_dtypes` (set by `detect_gguf_proj_dtypes`) can hold quantized per-projection dtypes even when the file majority is F16.

In that case the export takes the f32 dequantize path (writes f32 weights.bin), but the codegen sees `proj_dtypes` and emits Q4_K/Q6_K kernels with raw-byte slice sizes. Layout mismatch → garbage logits.

## Repro

A synthetic GGUF where ~50% of projection tensors are F16 and ~50% are Q4_K. `detect_gguf_dtype` returns F16; `detect_gguf_proj_dtypes` returns a mix with quantized dtypes for some categories. `forge compile` + `forge export-weights` produces an inconsistent pair.

No real GGUF in the wild has this exact mix today — but `detect_gguf_dtype`'s fallback semantics are loose enough that this regressing on a future format isn't far-fetched.

## Fix sketch

Gate the outer branch on whether **any** projection in `effective_proj_dtypes()` is byte-storage (Q8_0/Q4_0/Q4_K/Q6_K), not on `config.dtype`:

```rust
let pdt = config.effective_proj_dtypes();
let any_quantized = pdt.uses(DType::Q8_0)
    || pdt.uses(DType::Q4_0)
    || pdt.uses(DType::Q4_K)
    || pdt.uses(DType::Q6_K);
if any_quantized { /* quantized export */ } else { /* f32 export */ }
```

The `quant_label` computation already handles the mixed case — same pattern.

## Test

Add a unit test with a synthetic GGUF that has F16-majority + a few Q4_K/Q6_K projections, assert that `cmd_export_weights_impl` follows the quantized path.

**Comment by sauravpanda** [comment]
Fixed in v0.9.14 (534cadb): export path now gates on effective_proj_dtypes().uses(Qx) instead of config.dtype.
